### PR TITLE
ETL upgrade (omero56 update/ required ETL features)

### PR DESCRIPTION
**Pull request by luiskuhn**

### CODE REVIEW

File: drop-boxes/register-omero-metadata/image_registration_process.py [hunk 1]
```diff
@@ -5,16 +5,18 @@
...
 import csv
 from subprocess import Popen, PIPE
+import xml.etree.ElementTree as ET
+import os
 
 barcode_pattern = re.compile('Q[a-zA-Z0-9]{4}[0-9]{3}[A-Z][a-zA-Z0-9]')
-conda_home_path = "/home/qeana10/miniconda2/"
+conda_home_path = "/home/qeana10/miniconda3/"
 omero_lib_path = "/home/qeana10/openbis/servers/core-plugins/QBIC/1/dss/drop-boxes/register-omero-metadata/OMERO.py-5.4.10-ice36-b105"
 etl_home_path = "/home/qeana10/openbis/servers/core-plugins/QBIC/1/dss/drop-boxes/register-omero-metadata/"
```
Comment (Steffengreiner) [inline drop-boxes/register-omero-metadata/image_registration_process.py]: Maybe NIT: I think hardcoding the paths directly is dangerous and unnecessary and would advise to rather rely on Environment Variables or parameters during startup of the python script.

File: drop-boxes/register-omero-metadata/image_registration_process.py [hunk 2]
```diff
@@ -41,45 +43,26 @@ def __init__(self, transaction, env_name="omero_env_0", project_code="", sample_
...
         self._init_cmd_list.append('conda activate ' + env_name)
 
-        self._init_cmd_list.append('export OMERO_PREFIX=' + self._omero_path)
-        self._init_cmd_list.append('export PYTHONPATH=$PYTHONPATH:$OMERO_PREFIX/lib/python')
+        #self._init_cmd_list.append('export OMERO_PREFIX=' + self._omero_path)
+        #self._init_cmd_list.append('export PYTHONPATH=$PYTHONPATH:$OMERO_PREFIX/lib/python')
         
         # now use the omero-importer app packaged in the conda env
-        self._init_cmd_list.append('export PATH=$PATH:' + self._conda_path + 'envs/' + env_name + '/bin')
+        #self._init_cmd_list.append('export PATH=$PATH:' + self._conda_path + 'envs/' + env_name + '/bin')
```
Comment (Steffengreiner) [inline drop-boxes/register-omero-metadata/image_registration_process.py]: Remove unused code?

File: drop-boxes/register-omero-metadata/image_registration_process.py [hunk 3]
```diff
@@ -89,43 +72,115 @@ def requestOmeroDatasetId(self, project_code=None, sample_code=None):
...
+        out, err = self.executeCommandList(cmd_list)
 
-        process = Popen( "/bin/bash", shell=False, universal_newlines=True, stdin=PIPE, stdout=PIPE, stderr=PIPE )
-        out, err = process.communicate( commands )
-
-        ds_id = str(out)
+        ds_id = -1
+        output_xml = ET.fromstring(str(out))
+        for output_item in output_xml:
+            # print "---> " + str(output_item.tag) + " - " + str(output_item.attrib)
```
Comment (Steffengreiner) [inline drop-boxes/register-omero-metadata/image_registration_process.py]:
```suggestion
```

File: drop-boxes/register-omero-metadata/image_registration_process.py [hunk 4]
```diff
@@ -89,43 +72,115 @@ def requestOmeroDatasetId(self, project_code=None, sample_code=None):
...
 
-        commands = ""
-        for cmd in cmd_list:
-            commands = commands + cmd + "\n"
+        out, err = self.executeCommandList(cmd_list)
 
-        process = Popen( "/bin/bash", shell=False, universal_newlines=True, stdin=PIPE, stdout=PIPE, stderr=PIPE )
-        out, err = process.communicate( commands )
+        id_list = []
+        output_xml = ET.fromstring(str(out))
```
Comment (Steffengreiner) [inline drop-boxes/register-omero-metadata/image_registration_process.py:84]: Very NIT: Since your're importing a package not a constant i'd expect the import name to be lowercase
```suggestion
        output_xml = elementTree.fromstring(str(out))
```

File: drop-boxes/register-omero-metadata/image_registration_process.py [hunk 5]
```diff
@@ -89,43 +72,115 @@ def requestOmeroDatasetId(self, project_code=None, sample_code=None):
...
+        for output_item in output_xml:
+            img_id = output_item.attrib["id"]
```
Comment (Steffengreiner) [inline drop-boxes/register-omero-metadata/image_registration_process.py:86]: This will throw an unhandled error if the output_item does not have the "id" attribute?

File: drop-boxes/register-omero-metadata/image_registration_process.py [hunk 6]
```diff
@@ -89,43 +72,115 @@ def requestOmeroDatasetId(self, project_code=None, sample_code=None):
...
+            ds_id = int(output_item.attrib["id"])
+            break
 
         return ds_id
 
     def registerImageFileInOmero(self, file_path, dataset_id):
 
         cmd_list = list(self._init_cmd_list)
-        cmd_list.append( "python backendinterface.py -f " + file_path + " -d " + str(dataset_id) )
+        cmd_list.append( "omero-bifrost push img-file " + file_path + " " + str(dataset_id) + " --to-xml" )
```
Comment (Steffengreiner) [inline drop-boxes/register-omero-metadata/image_registration_process.py:79]: The --to-xml parameter is reused in multiple methods. 
I'd advise to make this a global constant since you reuse it in a multiple methods

File: drop-boxes/register-omero-metadata/image_registration_process.py [hunk 7]
```diff
@@ -89,43 +72,115 @@ def requestOmeroDatasetId(self, project_code=None, sample_code=None):
...
+    def registerOmeroKeyValuePairs(self, image_id, property_map):
+        """Registers the property map as key-value pairs in the OMERO server.
+        """
         
+        cmd_list = list(self._init_cmd_list)
+
+        key_value_str = ""
+        for key in property_map.keys(): 
+            key_value_str = key_value_str + "--kv-pair " + str(key) + ":" + str(property_map[key]) + " "
+        key_value_str = key_value_str[:len(key_value_str)-2] #remove last two chars
```
Comment (Steffengreiner) [inline drop-boxes/register-omero-metadata/image_registration_process.py]: What's the purpose of removing the last two chars here? I think a small line explaining your logic would help 👍 
```suggestion
        for key in property_map.keys(): 
            key_value_str = key_value_str + "--kv-pair " + str(key) + ":" + str(property_map[key]) + " "
        key_value_str = key_value_str[:len(key_value_str)-2] #remove last two chars
```

File: drop-boxes/register-omero-metadata/image_registration_process.py [hunk 8]
```diff
@@ -89,43 +72,115 @@ def requestOmeroDatasetId(self, project_code=None, sample_code=None):
...
+
+        cmd_list.append( "omero-bifrost push key-value " + str(image_id) + " " + key_value_str )
+
+        out, err = self.executeCommandList(cmd_list)
+
+        return 0
+
+    # TODO: clean following functions
```
Comment (Steffengreiner) [inline drop-boxes/register-omero-metadata/image_registration_process.py]: Address or remove ToDo
```suggestion
```

File: drop-boxes/register-omero-metadata/image_registration_process.py [hunk 9]
```diff
@@ -89,43 +72,115 @@ def requestOmeroDatasetId(self, project_code=None, sample_code=None):
...
+
+    def scanImageFolder(self, image_folder_path):
+        """Scans folder for image files and returns list of file paths
+        """
```
Comment (Steffengreiner) [inline drop-boxes/register-omero-metadata/image_registration_process.py]: I think this description is not true since this method is actually looking for '.tsv' files?

File: drop-boxes/register-omero-metadata/image_registration_process.py [hunk 10]
```diff
@@ -89,43 +72,115 @@ def requestOmeroDatasetId(self, project_code=None, sample_code=None):
...
+
+        cmd_list.append( "omero-bifrost push key-value " + str(image_id) + " " + key_value_str )
+
+        out, err = self.executeCommandList(cmd_list)
+
+        return 0
+
+    # TODO: clean following functions
+
+    def scanImageFolder(self, image_folder_path):
```
Comment (Steffengreiner) [inline drop-boxes/register-omero-metadata/image_registration_process.py]: I think a more explicit name or description would help in this method (What is the imagefolder scanned for?) 
```suggestion
    def scanImageFolder(self, image_folder_path):
```

File: drop-boxes/register-omero-metadata/image_registration_process.py [hunk 11]
```diff
@@ -89,43 +72,115 @@ def requestOmeroDatasetId(self, project_code=None, sample_code=None):
...
+        """Scans folder for image files and returns list of file paths
+        """
+
+        for root, subFolders, files in os.walk(image_folder_path):
+            for f in files:
+                stem, ext = os.path.splitext(f)
+                if ext.lower() =='.tsv':
+                    with open(os.path.join(root, f), 'U') as fh: metadataFileContent = fh.readlines()
```
Comment (Steffengreiner) [inline drop-boxes/register-omero-metadata/image_registration_process.py]: There seems to be no error handling here? 
What happens if there is an unrelated tsv file in the folder or the parsed paths are invalid?

File: drop-boxes/register-omero-metadata/image_registration_process.py [hunk 12]
```diff
@@ -89,43 +72,115 @@ def requestOmeroDatasetId(self, project_code=None, sample_code=None):
...
+        return metadataFileContent
 
     def triggerOMETiffConversion(self):
         pass
 
-    #ToDo Check if Metadata file is provided as defined
+    #TODO: Check if Metadata file is provided as defined
```
Comment (Steffengreiner) [inline drop-boxes/register-omero-metadata/image_registration_process.py]: Adress or remove ToDo 
```suggestion
```

File: drop-boxes/register-omero-metadata/image_registration_process.py [hunk 13]
```diff
@@ -144,30 +199,63 @@ def extractMetadataFromTSV(self, tsv_file_path):
...
-        cmd_list = list(self._init_cmd_list)
+    def fetchOpenBisSampleCode(self):
+        found = barcode_pattern.findall(self._incoming_file_name)
+        if len(found) == 0:
+            raise SampleCodeError(self._incoming_file_name, "Sample code does not match the QBiC sample code schema.")
+        self._sample_code = found[0]
+        if self._isValidSampleCode(self._sample_code):
+            self._project_code = self._sample_code[:5]
+        else:
+            raise SampleCodeError(self._sample_code, "The sample code seems to be invalid, the checksum could not be confirmed.")
```
Comment (Steffengreiner) [inline drop-boxes/register-omero-metadata/image_registration_process.py:180]: Nice error handling 👍

File: drop-boxes/register-omero-metadata/register-omero.py [hunk 14]
```diff
@@ -59,7 +60,16 @@
 INCOMING_DATE_FORMAT = '%d.%m.%Y'
 OPENBIS_DATE_FORMAT = '%Y-%m-%d'
 
-PROPPERTY_FILTER_LIST = ["IMAGE_FILENAME", "INSTRUMENT_USER", "IMAGING_DATE"]
+#PROPPERTY_FILTER_LIST = ["IMAGE_FILENAME", "INSTRUMENT_USER", "IMAGING_DATE"]
```
Comment (Steffengreiner) [inline drop-boxes/register-omero-metadata/register-omero.py]: Remove outdated filter list 
```suggestion
```

File: drop-boxes/register-omero-metadata/register-omero.py [hunk 15]
```diff
@@ -59,7 +60,16 @@
...
+PROPPERTY_FILTER_LIST = ["IMAGE_FOLDER_PATH", "INSTRUMENT_USER", "IMAGING_DATE", "SAMPLE_ID", "OMERO_TAGS", "ETL_TAG"]
```
Comment (Steffengreiner) [inline drop-boxes/register-omero-metadata/register-omero.py:64]: Typo? 
```suggestion
PROPERTY_FILTER_LIST = ["IMAGE_FOLDER_PATH", "INSTRUMENT_USER", "IMAGING_DATE", "SAMPLE_ID", "OMERO_TAGS", "ETL_TAG"]
```

File: drop-boxes/register-omero-metadata/register-omero.py [hunk 16]
```diff
@@ -59,7 +60,16 @@
...
+
+def log_print(msg_string):
+	import time
```
Comment (Steffengreiner) [inline drop-boxes/register-omero-metadata/register-omero.py]: import should be on top of the class not within methods

File: drop-boxes/register-omero-metadata/register-omero.py [hunk 17]
```diff
@@ -59,7 +60,16 @@
...
+
+	dt_string = time.strftime('%Y-%m-%d %H:%M:%S')
```
Comment (Steffengreiner) [inline drop-boxes/register-omero-metadata/register-omero.py:70]: Why not use the OpenBis DateFormat Variable here? 
```suggestion
	dt_string = time.strftime('%Y-%m-%d %H:%M:%S')
```

File: drop-boxes/register-omero-metadata/register-omero.py [hunk 18]
```diff
@@ -92,7 +102,7 @@ def createNewImagingExperiment(tr, space, project, properties, existing_ids):
 def createNewImagingRun(tr, base_sample, exp, omero_image_ids, run_offset, properties):
 	IMG_RUN_PREFIX = "IMG"
 	IMG_RUN_TYPE = "Q_BMI_GENERIC_IMAGING_RUN"
-	IMG_RUN_OMERO_PROPERTY_CODE = "Q_OMERO_IDS"
+	# IMG_RUN_OMERO_PROPERTY_CODE = "Q_OMERO_IDS"
```
Comment (Steffengreiner) [inline drop-boxes/register-omero-metadata/register-omero.py]: Remove outdated line
```suggestion
```

File: drop-boxes/register-omero-metadata/register-omero.py [hunk 19]
```diff
@@ -108,7 +118,7 @@ def createNewImagingRun(tr, base_sample, exp, omero_image_ids, run_offset, prope
 	img_run = tr.createNewSample(new_sample_id_with_offset, IMG_RUN_TYPE)
 	img_run.setParentSampleIdentifiers([base_sample.getSampleIdentifier()])
 	img_run.setExperiment(exp)
-	img_run.setPropertyValue(IMG_RUN_OMERO_PROPERTY_CODE, '\n'.join(omero_image_ids))
+	# img_run.setPropertyValue(IMG_RUN_OMERO_PROPERTY_CODE, '\n'.join(omero_image_ids))
```
Comment (Steffengreiner) [inline drop-boxes/register-omero-metadata/register-omero.py]: Remove outdated line
```suggestion
```

File: drop-boxes/register-omero-metadata/register-omero.py [hunk 20]
```diff
@@ -119,6 +129,9 @@ def createNewImagingRun(tr, base_sample, exp, omero_image_ids, run_offset, prope
 def getFileFromLine(line):
 	return line.split("\t")[0]
 
+def getImageFolderPathFromLine(line):
+	return line.split("\t")[0]
```
Comment (Steffengreiner) [inline drop-boxes/register-omero-metadata/register-omero.py]: These methods do the same thing, I'd recommend to diversify them by e.g. checking if the line is a valid path for a folder or file?

File: drop-boxes/register-omero-metadata/register-omero.py [hunk 21]
```diff
@@ -178,7 +191,8 @@ def validatePropertyNames(property_names):
 	"""
 
 	# fast validation without parser object.
-	required_names = ["IMAGE_FILENAME", "IMAGING_MODALITY", "IMAGED_TISSUE", "INSTRUMENT_MANUFACTURER", "INSTRUMENT_USER", "IMAGING_DATE"]
+	# required_names = ["IMAGE_FILENAME", "IMAGING_MODALITY", "IMAGED_TISSUE", "INSTRUMENT_MANUFACTURER", "INSTRUMENT_USER", "IMAGING_DATE"]
```
Comment (Steffengreiner) [inline drop-boxes/register-omero-metadata/register-omero.py]: Remove outdated code block 
```suggestion
```

File: drop-boxes/register-omero-metadata/register-omero.py [hunk 22]
```diff
@@ -194,10 +208,14 @@ def getPropertyMap(line, property_names):
 	property_values = line.split("\t")
 
 	for i in range(0, len(property_names)): #do not exclude first col (filename), the schema checks for it
-		##remove trailing newline, and replace space with underscore
+		## remove trailing newline, and replace space with underscore
 		name = property_names[i].rstrip('\n').replace(" ", "_")
```
Comment (Steffengreiner) [inline drop-boxes/register-omero-metadata/register-omero.py:211]: Nit: You could think about moving the rstrip and replace calls into a seperate method with documentation since it's applied multiple times and it's unclear why it's necessary here.

File: drop-boxes/register-omero-metadata/register-omero.py [hunk 23]
```diff
@@ -194,10 +208,14 @@ def getPropertyMap(line, property_names):
...
 		value = property_values[i].rstrip('\n').replace(" ", "_")
 
+		# look for placeholder symbol to skip property ("*")
+		if value == "*":
+			continue
```
Comment (Steffengreiner) [inline drop-boxes/register-omero-metadata/register-omero.py]: NIT: I'd suggest making this into a dedicated variable with a small line of explanation where the placeholder is applicable
```suggestion
		# look for placeholder symbol to skip property ("*")
		if value == "*":
			continue
```

File: drop-boxes/register-omero-metadata/register-omero.py [hunk 24]
```diff
@@ -325,39 +344,85 @@ def process(transaction):
...
+
+		log_print("++++++++++++++++++++++++++++++")
+		log_print("Starting table entry iteration")
+
+		# Get modality and other metadata from tsv.
+		# Additional metadata is provided in an own metadata TSV file. 
+		# We extract the metadata from this file.
 		properties = {}
+		properties = getPropertyMap(line, property_names)
+		# log_print("Line props: " + str(properties))
```
Comment (Steffengreiner) [inline drop-boxes/register-omero-metadata/register-omero.py]: Remove outdated line 
```suggestion
```

File: drop-boxes/register-omero-metadata/register-omero.py [hunk 25]
```diff
@@ -325,39 +344,85 @@ def process(transaction):
...
+			registrationProcess = defaultRegistrationProcess
 		
-		# Retrieve the image file name, please no whitespace characters in filename!
-		fileName = getFileFromLine(line)
+		# # Retrieve the image file name, please no whitespace characters in filename!
+		# fileName = getFileFromLine(line)
+		# # Due to the datahandler we need to add another subfolder of the same name to the path
+		# imageFolder = os.path.join(incomingPath, folderName)
+		# imageFile = os.path.join(imageFolder, fileName)
+		# log_print("New incoming image file for OMERO registration:\t" + imageFile)
```
Comment (Steffengreiner) [inline drop-boxes/register-omero-metadata/register-omero.py]: Remove outdated code block
```suggestion
```

File: drop-boxes/register-omero-metadata/register-omero.py [hunk 26]
```diff
@@ -325,39 +344,85 @@ def process(transaction):
...
+		else:
+			omero_dataset_id = default_omero_dataset_id
+		log_print("Iteration omero dataset id: " + str(omero_dataset_id))
+
 
 		# 4. After we have received the omero dataset id, we know where to attach the image to
 		# in OMERO. We pass the omero dataset id and trigger the image registration process in OMERO.
-		omero_image_ids = registrationProcess.registerImageFileInOmero(imageFile, omero_dataset_id)
-		print "Created OMERO image identifiers:\t" + str(omero_image_ids)
+		# omero_image_ids = registrationProcess.registerImageFileInOmero(imageFile, omero_dataset_id)
```
Comment (Steffengreiner) [inline drop-boxes/register-omero-metadata/register-omero.py]: Remove outdatedl ine 
```suggestion
```

File: drop-boxes/register-omero-metadata/register-omero.py [hunk 27]
```diff
@@ -325,39 +344,85 @@ def process(transaction):
...
 		if omero_failed:
 			raise ValueError("Omero did not return expected image ids.")
 
-		# 5. Additional metadata is provided in an own metadata TSV file. 
-		# We extract the metadata from this file.
-		properties = getPropertyMap(line, property_names)
-
 		# 5.1 Validate metadata for image file
-		ImagingMetadataValidator.validateImagingProperties(getValidationMap(properties))
+		# ImagingMetadataValidator.validateImagingProperties(getValidationMap(properties))
```
Comment (Steffengreiner) [inline drop-boxes/register-omero-metadata/register-omero.py:425]: Remove outdatedl ine
```suggestion
```

File: drop-boxes/register-omero-metadata/register-omero.py [hunk 28]
```diff
@@ -325,39 +344,85 @@ def process(transaction):
...
+		# Annotate with metadata, using OMERO key-value pairs and tags 
+		# one file can have many images, iterate over all img ids
 		for img_id in omero_image_ids:
 			registrationProcess.registerOmeroKeyValuePairs(img_id, filterOmeroPropertyMap(properties, PROPPERTY_FILTER_LIST))
 
+		if "OMERO_TAGS" in properties.keys():
+			tag_list = properties["OMERO_TAGS"].split(",")
+			for tag_value in tag_list:
+				for img_id in omero_image_ids:
+					registrationProcess.tagOmeroImage(img_id, tag_value)
```
Comment (Steffengreiner) [inline drop-boxes/register-omero-metadata/register-omero.py:436]: What happens here if OMERO_TAGS is not in properties.keys()?

File: drop-boxes/register-omero-metadata/register-omero.py [hunk 29]
```diff
@@ -379,3 +444,5 @@ def process(transaction):
 
 		# 7. Last but not least we create the open science file format for images which is
 		# OMERO-Tiff and store it in OMERO next to the proprierary vendor format.
```
Comment (Steffengreiner) [inline drop-boxes/register-omero-metadata/register-omero.py]: Outdated description since the only thing that's happening here is a log_print?

**Review by Steffengreiner — CHANGES_REQUESTED** [review]
Heya @luiskuhn Thanks for your awesome contribution. I found a lot of outdated code segments and have some questions concerning error handling. I'd approve the PR once these questions are answered

File: drop-boxes/register-omero-metadata/image_registration_process.py (hunk 5 above)
Comment (luiskuhn) [reply drop-boxes/register-omero-metadata/image_registration_process.py:86]: Good point, it will break in such case. I will patch that

File: drop-boxes/register-omero-metadata/register-omero.py (hunk 28 above)
Comment (luiskuhn) [reply drop-boxes/register-omero-metadata/register-omero.py:436]: If there was no OMERO_TAG col. in the tsv file, then don't tag the images. The tags should be a string of tag separated by commas, I should explain this properly in a comment it in the code, and in the readme doc.

File: drop-boxes/register-omero-metadata/image_registration_process.py (hunk 1 above)
Comment (luiskuhn) [reply drop-boxes/register-omero-metadata/image_registration_process.py]: Good point, I plan to refactor that, but atm I am not what is the best way to do it using the factory pattern that was introduced. I will add a TODO comment here

Comment (luiskuhn) [reply drop-boxes/register-omero-metadata/image_registration_process.py]: I also added those parameters in the `ImageRegistrationProcess` constructor:
https://github.com/qbicsoftware/etl-scripts/blob/f28aad21d189ca4d0148fc5245beceeec9135a46/drop-boxes/register-omero-metadata/image_registration_process.py#L19

File: drop-boxes/register-omero-metadata/image_registration_process.py (hunk 2 above)
Comment (luiskuhn) [reply drop-boxes/register-omero-metadata/image_registration_process.py]: done

File: drop-boxes/register-omero-metadata/image_registration_process.py (hunk 3 above)
Comment (luiskuhn) [reply drop-boxes/register-omero-metadata/image_registration_process.py]: done

File: drop-boxes/register-omero-metadata/image_registration_process.py (hunk 5 above)
Comment (luiskuhn) [reply drop-boxes/register-omero-metadata/image_registration_process.py:86]: Ok I double-checked, that xml structure is the standard output of the omero-bifrost tool, so it will always have an id attribute. It is part of the specification of the omero-bifrost tool, so we can rely in that all xml output will have it.

File: drop-boxes/register-omero-metadata/image_registration_process.py (hunk 6 above)
Comment (luiskuhn) [reply drop-boxes/register-omero-metadata/image_registration_process.py:79]: Right, atm I am not sure how to parametrize that in a way that makes sense, it is just a flag passed to omero-bifrost, and the aim is to always communicate with xml, so it wont change much.

File: drop-boxes/register-omero-metadata/image_registration_process.py (hunk 7 above)
Comment (luiskuhn) [reply drop-boxes/register-omero-metadata/image_registration_process.py]: Good point, thank you. The idea was to remove the last white space, I added a comment. But you also found a bug :) its fixed now

File: drop-boxes/register-omero-metadata/image_registration_process.py (hunk 8 above)
Comment (luiskuhn) [reply drop-boxes/register-omero-metadata/image_registration_process.py]: Will do, some TODO comments I would like to keep since they point to the places where we want to make further modifications in the coming releases. But I will try to minimize them

File: drop-boxes/register-omero-metadata/image_registration_process.py (hunk 9 above)
Comment (luiskuhn) [reply drop-boxes/register-omero-metadata/image_registration_process.py]: Good point, I will change the description

File: drop-boxes/register-omero-metadata/image_registration_process.py (hunk 10 above)
Comment (luiskuhn) [reply drop-boxes/register-omero-metadata/image_registration_process.py]: good point will rename it

File: drop-boxes/register-omero-metadata/image_registration_process.py (hunk 11 above)
Comment (luiskuhn) [reply drop-boxes/register-omero-metadata/image_registration_process.py]: Just realized we don't need this function, will delete it

File: drop-boxes/register-omero-metadata/image_registration_process.py (hunk 12 above)
Comment (luiskuhn) [reply drop-boxes/register-omero-metadata/image_registration_process.py]: removed

File: drop-boxes/register-omero-metadata/register-omero.py (hunk 15 above)
Comment (luiskuhn) [reply drop-boxes/register-omero-metadata/register-omero.py:64]: That should be fine, these new properties should be filtered as well

File: drop-boxes/register-omero-metadata/register-omero.py (hunk 16 above)
Comment (luiskuhn) [reply drop-boxes/register-omero-metadata/register-omero.py]: You can also do it inside functions to keep your namespace clean :)
but I can bring it to top

Comment (luiskuhn) [reply drop-boxes/register-omero-metadata/register-omero.py]: done

File: drop-boxes/register-omero-metadata/register-omero.py (hunk 17 above)
Comment (luiskuhn) [reply drop-boxes/register-omero-metadata/register-omero.py:70]: This is just a format to make the logging clearer, not sure if we need to stay consistent with the OpenBis DateFormat Variable

File: drop-boxes/register-omero-metadata/register-omero.py (hunk 18 above)
Comment (luiskuhn) [reply drop-boxes/register-omero-metadata/register-omero.py]: done

File: drop-boxes/register-omero-metadata/register-omero.py (hunk 19 above)
Comment (luiskuhn) [reply drop-boxes/register-omero-metadata/register-omero.py]: done

File: drop-boxes/register-omero-metadata/register-omero.py (hunk 20 above)
Comment (luiskuhn) [reply drop-boxes/register-omero-metadata/register-omero.py]: Good point, `getFileFromLine()` was redundant, I removed it

File: drop-boxes/register-omero-metadata/register-omero.py (hunk 22 above)
Comment (luiskuhn) [reply drop-boxes/register-omero-metadata/register-omero.py:211]: Ok, I can think of making it more expressive, I will add a note here

File: drop-boxes/register-omero-metadata/register-omero.py (hunk 23 above)
Comment (luiskuhn) [reply drop-boxes/register-omero-metadata/register-omero.py]: Good point, I added a global `PROPPERTY_PLACEHOLDER` constant, with a line explaining the meaning

File: drop-boxes/register-omero-metadata/register-omero.py (hunk 24 above)
Comment (luiskuhn) [reply drop-boxes/register-omero-metadata/register-omero.py]: done

File: drop-boxes/register-omero-metadata/register-omero.py (hunk 25 above)
Comment (luiskuhn) [reply drop-boxes/register-omero-metadata/register-omero.py]: done

File: drop-boxes/register-omero-metadata/register-omero.py (hunk 26 above)
Comment (luiskuhn) [reply drop-boxes/register-omero-metadata/register-omero.py]: done

File: drop-boxes/register-omero-metadata/register-omero.py (hunk 27 above)
Comment (luiskuhn) [reply drop-boxes/register-omero-metadata/register-omero.py:425]: This line I would like to keep, since here I am just temporally disabled metadata validation, pending REMBI alignment and and some ETL logic modifications, but this will not make this cut for the release. This REMBI alignment of ontology vocabularies needs to be done properly, so I will just patch this later once we update the metadata model properly.

File: drop-boxes/register-omero-metadata/register-omero.py (hunk 29 above)
Comment (luiskuhn) [reply drop-boxes/register-omero-metadata/register-omero.py]: Its removed

File: drop-boxes/register-omero-metadata/image_registration_process.py (hunk 1 above)
Comment (Steffengreiner) [reply drop-boxes/register-omero-metadata/image_registration_process.py]: Perfect 👌

File: drop-boxes/register-omero-metadata/image_registration_process.py (hunk 5 above)
Comment (Steffengreiner) [reply drop-boxes/register-omero-metadata/image_registration_process.py:86]: Thanks for double checking! I'd still recommend having an exception here just to be sure since it can provide more detail than a nullpointer exception 👍

File: drop-boxes/register-omero-metadata/image_registration_process.py (hunk 6 above)
Comment (Steffengreiner) [reply drop-boxes/register-omero-metadata/image_registration_process.py:79]: One possibility could be to use a dedicated argument parser in the future such as https://docs.python.org/3/library/argparse.html

File: drop-boxes/register-omero-metadata/image_registration_process.py (hunk 7 above)
Comment (Steffengreiner) [reply drop-boxes/register-omero-metadata/image_registration_process.py]: To remove whitespaces i'd recommend to use the python native [strip method] in the future (https://stackoverflow.com/questions/1185524/how-do-i-trim-whitespace)

File: drop-boxes/register-omero-metadata/image_registration_process.py (hunk 9 above)
Comment (Steffengreiner) [reply drop-boxes/register-omero-metadata/image_registration_process.py]: Nice 👍

File: drop-boxes/register-omero-metadata/image_registration_process.py (hunk 10 above)
Comment (Steffengreiner) [reply drop-boxes/register-omero-metadata/image_registration_process.py]: Nice 👍

File: drop-boxes/register-omero-metadata/image_registration_process.py (hunk 11 above)
Comment (Steffengreiner) [reply drop-boxes/register-omero-metadata/image_registration_process.py]: Even better 🥳

File: drop-boxes/register-omero-metadata/register-omero.py (hunk 15 above)
Comment (Steffengreiner) [reply drop-boxes/register-omero-metadata/register-omero.py:64]: I just meant the variable name since it's ProPPerty instead of ProPerty 👍

File: drop-boxes/register-omero-metadata/register-omero.py (hunk 23 above)
Comment (Steffengreiner) [reply drop-boxes/register-omero-metadata/register-omero.py]: Nice 🆗

File: drop-boxes/register-omero-metadata/register-omero.py (hunk 27 above)
Comment (Steffengreiner) [reply drop-boxes/register-omero-metadata/register-omero.py:425]: That's fine however I'd then recommend a ToDo here instead of having the line duplicated and commented out 🤔

File: drop-boxes/register-omero-metadata/register-omero.py (hunk 28 above)
Comment (Steffengreiner) [reply drop-boxes/register-omero-metadata/register-omero.py:436]: That would be perfect thank you 🙏

File: drop-boxes/register-omero-metadata/image_registration_process.py (hunk 8 above)
Comment (Steffengreiner) [reply drop-boxes/register-omero-metadata/image_registration_process.py]: That's completely valid, I'm just bringing it up to double check their valiidity

**Review by Steffengreiner — DISMISSED** [review]
Nicely done!

**Review by Steffengreiner — DISMISSED** [review]
Thanks for the documentation update 👌

**Comment by luiskuhn** [comment]
Documentation is done, I think its ready to go :)

**Review by Steffengreiner — APPROVED** [review]
Ship it 🚢
